pkg/network/ovn_kubernetes.go

```diff
@@ -633,18 +614,20 @@ func shouldRenderIPsec(conf *operv1.OVNKubernetesConfig, bootstrapResult *bootst
 
 	mode := GetIPsecMode(conf)
 
-	// We render the ipsec deployment if IPsec is already active in OVN
-	// or if EW IPsec config is enabled.
-	renderIPsecDaemonSet = isOVNIPsecActive || mode == operv1.IPsecModeFull
+	// We render the ipsec deployment if IPsec is already active in OVN or if EW IPsec config is enabled.
+	// When it's not a hypershift hosted cluster we also need to ensure machine configs present though
+	// isOVNIPsecActive is true.
+	isCNOIPsecMachineConfigPresent := isCNOIPsecMachineConfigPresent(bootstrapResult.Infra)
+	isUserDefinedIPsecMachineConfigPresent := isUserDefinedIPsecMachineConfigPresent(bootstrapResult.Infra)
+	renderIPsecDaemonSet = (isOVNIPsecActive && (isCNOIPsecMachineConfigPresent || isUserDefinedIPsecMachineConfigPresent ||
+		isHypershiftHostedCluster)) || mode == operv1.IPsecModeFull
```

Comment:
can you rather tweak isOVNIPsecActive above?
```
isOVNIPsecActiveOrRollingOut := bootstrapResult.OVN.IPsecUpdateStatus != nil && bootstrapResult.OVN.IPsecUpdateStatus.OVNRolloutInProgress
// when OVN is rolling out, OVN IPsec might be fully or partially active or inactive.  If MachineConfigs are not present, we know its inactive since we only stop rendering them once inactive.
isOVNIPsecActive := isOVNIPsecActiveOrRollingOut && (isCNOIPsecMachineConfigPresent || isUserDefinedIPsecMachineConfigPresent || isHypershiftHostedCluster)

```